generators/ci-cd/templates/.gitlab-ci.yml.ejs

```diff
@@ -152,18 +152,20 @@ before_script:
     <%_ } _%>
     - export MAVEN_USER_HOME=`pwd`/.maven
     <%_ if (!skipClient) { _%>
-        <%_ if (clientPackageManager === 'yarn') { _%>
+        <%_ if (!insideDocker) { _%>
+            <%_ if (clientPackageManager === 'yarn') { _%>
     - ./mvnw com.github.eirslett:frontend-maven-plugin:install-node-and-yarn -DnodeVersion=v<%= NODE_VERSION %> -DyarnVersion=v<%= YARN_VERSION %> -Dmaven.repo.local=$MAVEN_USER_HOME
-        <%_ } else if (clientPackageManager === 'npm') { _%>
+            <%_ } else if (clientPackageManager === 'npm') { _%>
     - ./mvnw com.github.eirslett:frontend-maven-plugin:install-node-and-npm -DnodeVersion=v<%= NODE_VERSION %> -DnpmVersion=<%= NPM_VERSION %> -Dmaven.repo.local=$MAVEN_USER_HOME
-        <%_ } _%>
+            <%_ } _%>
     - ./mvnw com.github.eirslett:frontend-maven-plugin:<%= clientPackageManager %> -Dmaven.repo.local=$MAVEN_USER_HOME
+        <%_ } _%>
     <%_ } _%>
 
 maven-compile:
     stage: build
     script:
-        - ./mvnw compile -Dmaven.repo.local=$MAVEN_USER_HOME
+        - ./mvnw compile -P-webpack -Dmaven.repo.local=$MAVEN_USER_HOME
```

Comment:
@pascalgrimaud, I think we can apply similar optimizations on current `travis` and `azure pipelines`. do you see an issue?

Reply:
@vishal423 : yes, we can give a try.

Not related: the only issue is our current daily builds, which are broken with angular since 2 days. I didn't have time to open a specific issue for that

Reply:
It seems all failures are related to e2e tests (including this this PR). I will try to spend some time today evening to debug further on that.

Reply:
@vishal423 yes the e2e tests are failing for some reasons. Did not get the chance to have a look at it. I am also curious about the reason for failing.

Reply:
I opened a ticket for that https://github.com/jhipster/generator-jhipster/issues/10371 as the e2e tests failures are not related to your PR